spec/queries/keywords_query_spec.rb

```diff
@@ -26,19 +26,132 @@
-      context 'given filtering with a word vpn and match at least 1' do
-        it 'returns a keyword' do
+    context 'when querying with result_url_contains' do
+      context 'when there are search urls containing the word vpn' do
+        it 'returns 2 keywords' do
+          result_urls = ['https://www.thetopvpn.com', 'https://www.nordvpn.com', 'https://www.vnexpress.net']
+          Fabricate.times(2, :keyword, result_urls: result_urls)
+
+          filter_params = { result_url_contains: 'vpn' }
+          keywords = described_class.new(Keyword, filter_params).call
+
+          expect(keywords.count).to eq 2
+        end
+      end
+
+      context 'when there is no urls containing the word vpn' do
+        it 'returns empty array for keywords' do
+          Fabricate(:keyword)
+
+          filter_params = { result_url_contains: 'vpn' }
+          keywords = described_class.new(Keyword, filter_params).call
+
+          expect(keywords).to be_empty
+        end
+      end
+    end
+
+    context 'when querying with adwords_url_contains and result_url_contains' do
+      context 'when there are urls containing the query words' do
+        it 'returns 2 keywords' do
+          ads_top_urls = ['https://www.thetopvpn.com', 'https://www.nordvpn.com', 'https://www.vnexpress.net']
+          result_urls = ['https://www.topgame.com']
+          Fabricate.times(2, :keyword, ads_top_urls: ads_top_urls, result_urls: result_urls)
+
+          filter_params = { adwords_url_contains: 'vpn', result_url_contains: 'game' }
+          keywords = described_class.new(Keyword, filter_params).call
+
+          expect(keywords.count).to eq 2
+        end
+      end
+
+      context 'when there is no urls containing the adwords_url_contains word' do
+        it 'returns empty array for keywords' do
+          ads_top_urls = ['https://www.thetop.com', 'https://www.nord.com', 'https://www.vnexpress.net']
+          result_urls = ['https://www.topgame.com']
+          Fabricate(:keyword, ads_top_urls: ads_top_urls, result_urls: result_urls)
+
+          filter_params = { adwords_url_contains: 'vpn', result_url_contains: 'game' }
+          keywords = described_class.new(Keyword, filter_params).call
+
+          expect(keywords).to be_empty
+        end
+      end
+
+      context 'when there is no urls containing the result_url_contains word' do
+        it 'returns empty array for keywords' do
+          ads_top_urls = ['https://www.vpn.com']
+          result_urls = ['https://www.top.com']
+          Fabricate(:keyword, ads_top_urls: ads_top_urls, result_urls: result_urls)
+
+          filter_params = { adwords_url_contains: 'vpn', result_url_contains: 'game' }
+          keywords = described_class.new(Keyword, filter_params).call
+
+          expect(keywords).to be_empty
+        end
+      end
+
+      context 'when there is no urls containing the query words' do
+        it 'returns empty array for keywords' do
```

Comment:
```suggestion
        it 'returns an empty array' do
```

Reply:
Fixed in https://github.com/nimblehq/ic-rails-phong-bliss/pull/78/commits/e91fb7286fde0e64ca3f24e6374149c1212aeebd